Source/Streamstone/EntityOperation.cs

```diff
@@ -132,25 +220,28 @@ public override EntityOperation Merge(EntityOperation other)
             {
                 if (other is Insert)
                     return other;
-                
+
                 if (other is Replace)
                     throw InvalidMerge(other);
 
                 if (other is Delete)
                     throw InvalidMerge(other);
 
-                throw new InvalidOperationException("Unsupported operation type: " + other.GetType());                   
+                if (other is InsertOrMerge)
```

Comment:
Looking at this, I think I might agree with @jkonecki that we should merge conditions with similar outcome, since we already have state transition table documented